app/controllers/warehouse_reports/incomes_controller.rb

```diff
@@ -33,6 +33,46 @@ def index
       end
     end
 
+    def headers_for_export
+      headers = [
+        'Warehouse Client ID',
+        'First Name',
+        'Last Name',
+      ] +
+          GrdaWarehouse::Hud::IncomeBenefit::SOURCES.keys.map { |source| "#{source.to_s.titleize} at Entry"  } +
+          GrdaWarehouse::Hud::IncomeBenefit::SOURCES.keys.map { |source| "#{source.to_s.titleize} at Update" } +
+      [
+        'Gender',
+        'Race',
+        'Ethnicity',
+      ]
+      headers = headers.excluding('First Name', 'Last Name') unless ::GrdaWarehouse::Config.get(:include_pii_in_detail_downloads)
+      headers
+    end
```

Comment:
```suggestion
    def headers_for_export
      headers = ['Warehouse Client ID']
      headers += ['First Name', 'Last Name'] if ::GrdaWarehouse::Config.get(:include_pii_in_detail_downloads)
      headers += [
          GrdaWarehouse::Hud::IncomeBenefit::SOURCES.keys.map { |source| "#{source.to_s.titleize} at Entry"  } +
          GrdaWarehouse::Hud::IncomeBenefit::SOURCES.keys.map { |source| "#{source.to_s.titleize} at Update" } +
      [
        'Gender',
        'Race',
        'Ethnicity',
      ]
      headers
    end
```

Reply:
Seems like we could do what you did in the rows_for_export method and just not adde them if we don't want them.